src/test/java/com/iexec/worker/compute/ComputeControllerTests.java

```diff
@@ -17,60 +17,215 @@
-        Assertions.assertEquals(200, response.getStatusCode().value());
+        Assertions.assertEquals(HttpStatus.OK.value(), response.getStatusCode().value());
     }
 
     @Test
     void shouldNotSendExitCauseForComputeComputeStageSinceNoCause() {
-        ResponseEntity<?> response =
-                computeController.sendExitCauseForGivenComputeStage(ComputeStage.PRE,
+        when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
+                .thenReturn(true);
+
+        final ResponseEntity<?> response =
+                computeController.sendExitCauseForGivenComputeStage(AUTH_HEADER,
+                        ComputeStage.PRE,
                         CHAIN_TASK_ID,
                         new ExitMessage());
-        Assertions.assertTrue(response.getStatusCode().is4xxClientError());
-        Assertions.assertEquals(400, response.getStatusCode().value());
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatusCode().value());
     }
 
     @Test
     void shouldNotSendExitCauseForComputeComputeStageSinceAlreadySet() {
-        ComputeStage stage = ComputeStage.PRE;
+        when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
+                .thenReturn(true);
+
+        final ComputeStage stage = ComputeStage.PRE;
         computeStageExitService.setExitCause(stage, CHAIN_TASK_ID, CAUSE);
-        //try to re set
-        ResponseEntity<?> response =
-                computeController.sendExitCauseForGivenComputeStage(stage,
-                        CHAIN_TASK_ID,
-                        new ExitMessage(CAUSE));
-        Assertions.assertTrue(response.getStatusCode().is2xxSuccessful());
-        Assertions.assertEquals(208, response.getStatusCode().value());
+
+        final ResponseEntity<Void> response = computeController.sendExitCauseForGivenComputeStage(
+                AUTH_HEADER,
+                stage,
+                CHAIN_TASK_ID,
+                new ExitMessage(CAUSE)
+        );
+
+        Assertions.assertEquals(HttpStatus.ALREADY_REPORTED.value(), response.getStatusCode().value());
+    }
+
+    @Test
+    void shouldReturnUnauthorizedWhenAuthFails() {
+        when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
+                .thenReturn(false);
+
+        final ResponseEntity<Void> response = computeController.sendExitCauseForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.PRE,
+                CHAIN_TASK_ID,
+                new ExitMessage(CAUSE)
+        );
+
+        Assertions.assertEquals(HttpStatus.UNAUTHORIZED.value(), response.getStatusCode().value());
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenWrongChainTaskId() {
+        when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
+                .thenThrow(new NoSuchElementException());
```

Comment:
I think we use more the following syntax in our tests
```suggestion
                .thenThrow(NoSuchElementException.class);
```

Reply:
There was a 2 occurences to fix in `ComputeControllerTests`.
One has been fixed, the other remains.